include/swift/Reflection/ReflectionContext.h

```diff
@@ -190,8 +193,112 @@ class ReflectionContext
 
     return true;
   }
-#endif // defined(__APPLE__) && defined(__MACH__)
-  
+#else // ELF platforms.
+  bool addImage(RemoteAddress ImageStart) {
+    auto Buf = this->getReader().readBytes(ImageStart, sizeof(Elf64_Ehdr));
+
+    // Read the header.
+    auto Hdr = reinterpret_cast<const Elf64_Ehdr *>(Buf.get());
+
+    if (!Hdr->checkMagic())
+      return false;
+
+    // From the header, grab informations about the section header table.
+    auto SectionHdrAddress = ImageStart.getAddressData() + Hdr->e_shoff;
```

Comment:
This is definitely broken with 32-bit images.  `offsetof(Elf_Ehdr, e_shoff)` is going to be 0x20 vs 0x28.  It also is a 4-byte vs 8-byte value.

Reply:
I understand this all is 64-bit specific. We can generalize to 32-bits later.